lib/widgets/new_dm_sheet.dart

```diff
@@ -0,0 +1,360 @@
+        contentPadding: EdgeInsets.zero,
+        hintText: hintText,
+        hintStyle: TextStyle(color: designVariables.labelSearchPrompt,
+          fontSize: 17, height: 22 / 17),
+        border: InputBorder.none));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+
+    return Container(
+      constraints: const BoxConstraints(
+        minHeight: 44,
+        maxHeight: 124),
+      padding: const EdgeInsets.symmetric(horizontal: 14, vertical: 11),
+      decoration: BoxDecoration(color: designVariables.bgSearchInput),
+      child: SingleChildScrollView(
+        reverse: true,
+        child: Row(children: [
+          Expanded(child: Wrap(
+            spacing: 6,
+            runSpacing: 4,
+            crossAxisAlignment: WrapCrossAlignment.center,
+            children: [
+              ...selectedUserIds.map((userId) => _SelectedUserChip(userId: userId)),
+              IntrinsicWidth(child: _buildSearchField(context)),
+            ])),
+        ])));
+  }
+}
+
+class _SelectedUserChip extends StatelessWidget {
+  const _SelectedUserChip({required this.userId});
+
+  final int userId;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final store = PerAccountStoreWidget.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    final user = store.getUser(userId);
+
+    return Container(
+      decoration: BoxDecoration(
+        color: designVariables.bgMenuButtonSelected,
+        borderRadius: BorderRadius.circular(3)),
+      child: Row(
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          Avatar(userId: userId, size: 22, borderRadius: 3),
+          Padding(
+            padding: const EdgeInsetsDirectional.fromSTEB(5, 3, 4, 3),
+            child: Text(user?.fullName ?? zulipLocalizations.unknownUserName,
+              style: TextStyle(
+                fontSize: 16,
+                height: 16 / 16,
+                color: designVariables.labelMenuButton))),
+        ]));
+  }
+}
+
+class _NewDmUserList extends StatelessWidget {
+  const _NewDmUserList({
+    required this.filteredUsers,
+    required this.selectedUserIds,
+    required this.onUserTapped,
+  });
+
+  final List<User> filteredUsers;
+  final Set<int> selectedUserIds;
+  final void Function(int userId) onUserTapped;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final store = PerAccountStoreWidget.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    if (filteredUsers.isEmpty) {
+      return Center(
+        child: Padding(
+          padding: const EdgeInsets.all(24),
+          child: Text(
+            zulipLocalizations.newDmSheetNoUsersFound,
+            style: TextStyle(
+              color: designVariables.labelMenuButton,
+              fontSize: 16))));
```

Comment:
This will need a `TODO(design)` comment since it is not a part of the Figma design.